Find the Largest Number in an Array #4 (new class + tests)

rafaello-rm opened the pull request.

rafaello-rm committed:
Find the Largest Number in an Array #4 (new class + tests)

kubapalasz commented on ConsoleApp3/Math.Services/LargestNumber.cs:
```diff
@@ -0,0 +1,37 @@
+            {
+                return result;
+            }
+            result = array[0];
```
@rafaello-rm , bardzo dobrze. 

Jak możesz to sobie poczytaj o **Linq**. 
Takie proste metody (i wiele więcej) ogólnie są już zaimplementowane...

Np.: https://app.pluralsight.com/library/courses/linq-fundamentals-csharp-6/table-of-contents

kubapalasz replied:
Przykładowo...
`return array.Min();` ;-)

rafaello-rm replied:
dzięki, chociaż muszę się przyznać, że nie do końca jestem zadowolony bo w testach jak tylko chciałem przekazać null'a albo nawet pustego string'a jako item w tablicy to się wysypywało (nie zgadzał mu się typ danych wejściowych) i nie bardzo wiedziałem jak sobie z tym poradzić.

kubapalasz commented on ConsoleApp3/Math.Services.Tests/LargestNumberTests.cs:
```diff
@@ -0,0 +1,23 @@
+        [Theory]
+        [InlineData(new[] {"1","5","7"}, "7")]
+        [InlineData(new[] {"54","1","0"}, "54")]        
+        [InlineData(new[] {"-5","6","-4"}, "6")]        
```
@rafaello-rm , dobrze, że wziąłeś też ujemne wartości pod uwagę ;-)

kubapalasz replied:
@rafaello-rm , dodałbym jeszcze tak jak w innym teście

[InlineData(null, "")]

rafaello-rm replied:
właśnie jak tylko chciałem przekazać null'a albo nawet pustego string'a jako item w tablicy to się wysypywało (nie zgadzał mu się typ danych wejściowych) i nie bardzo wiedziałem jak sobie z tym poradzić.

kubapalasz replied:
Null jako tablice można przekazać bo tablica to obiekt. 
Ale nie wiem, czy w tablicy int'ow można przekazać null, bo sam int nie  może być null. Bo to jest typ wartościowy 'value type'

kubapalasz commented on ConsoleApp3/ConsoleApp3/Program.cs:
```diff
@@ -1,22 +1,13 @@
-                pobieramyNastepnaLiczbe = (aktulanaLiczba != 123);
-            }
-
+            var wynik = LargestNumber.MaxFromArray(args);
```
@rafaello-rm , dziś postaram się na masterze trochę pozmieniać w tej części żeby łatwo szło dodawać kolejne zadania.

kubapalasz commented on ConsoleApp3/Math.Services/LargestNumber.cs:
```diff
@@ -0,0 +1,37 @@
+                return result = "";
+            }
+            int[] numbersArray = Array.ConvertAll(array, int.Parse);
+           return result = MaxFromArray(numbersArray).ToString();            
```
@rafaello-rm  w tej linijce zmienna `result` jest niepotrzebna.
Może to być zapisane tak:

 `return MaxFromArray(numbersArray).ToString();`

kubapalasz commented:
@rafaello-rm , jak przeczytasz uwagi to możesz zmergować ten PR do mastera

rafaello-rm committed:
Find the Largest Number in an Array #4 (added tests)

Commit:
#4 Updated method implementation + added couple of non trivial unit test cases

Commit:
#4 Merge remote-tracking branch 'origin/master' into feature/csv-sumator

Commit:
#4 Added example code of converter

Commit:
#4 Merge remote-tracking branch 'origin/master' into feature/csv-sumator

Commit:
#4 Merge remote-tracking branch 'origin/master' into feature/csv-sumator

# Conflicts:
#	ConsoleApp3/ConsoleApp3/ConsoleApp3.csproj
#	ConsoleApp3/ConsoleApp3/Program.cs

Commit:
#4 Added new case for Rafal

rafaello-rm committed:
#4 added input parameter support for Largest Number

rafaello-rm committed:
#4 missing char fix
